admin/client/src/containers/FormBuilderLoader/FormBuilderLoader.js

```diff
@@ -25,12 +43,55 @@ class FormBuilderLoader extends Component {
     }
   }
 
-  handleSubmitSuccess(result) {
-    this.props.schemaActions.setSchema(result);
+  /**
+   * Get server-side validation messages returned and display them on the form.
+   *
+   * @param state
+   * @returns {object}
+   */
+  getMessages(state) {
+    const messages = {};
 
-    if (this.props.onSubmitSuccess) {
-      this.props.onSubmitSuccess(result);
+    // only error messages are collected
+    // TODO define message type as standard "success", "info", "warning" and "danger"
```

Comment:
Yes, it's a bit scattered in terms of "error types" or "message types", I remember running into a case where some validation errors even return the validation test name rather than "error" as the type which convolutes what to look for even more.